geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/SchemaParser.scala

```diff
@@ -0,0 +1,452 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.kafka.confluent
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.locationtech.geomesa.utils.text.{DateParsing, WKBUtils, WKTUtils}
+import org.locationtech.jts.geom._
+import org.opengis.feature.simple.SimpleFeatureType
+
+import java.nio.ByteBuffer
+import java.time.format.DateTimeFormatter
+import java.util.{Date, Locale}
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.reflect.{ClassTag, classTag}
+import scala.util.control.NonFatal
+
+object SchemaParser {
```

Comment:
this class was copied from AvroSimpleFeatureTypeParser and probably doesn't need much/any review